compiler/rustc_const_eval/src/interpret/machine.rs

```diff
@@ -202,6 +203,7 @@ pub trait Machine<'tcx>: Sized {
         ecx: &mut InterpCx<'tcx, Self>,
         instance: ty::Instance<'tcx>,
         abi: ExternAbi,
+        _fnabi: &FnAbi<'tcx, Ty<'tcx>>,
```

Comment:
As discussed in the Miri issue, IMO this should *replace* the `ExternAbi` argument, not complement it.

Reply:
If we remove the ``ExternAbi`` here, we need to change how we handle ``check_abi_and_shim_symbol_clash`` in this PR too. I initially thought of splitting it into two pr:
1. Pass down ``FnAbi``
2. Handle ``check_abi_and_shim_symbol_clash`` and remove all unused ``ExternAbi``. 

But if this is preferred, I can do them in a single pr.

Reply:
I'd like to avoid growing the number of arguments, so yeah I figured this would be one PR.